lib/src/utils.dart

```diff
@@ -20,3 +22,13 @@ int rotl32(int val, int shift) {
   var modShift = shift & 31;
   return ((val << modShift) & mask32) | ((val & mask32) >> (32 - modShift));
 }
+
+/// Generates a cryptographically secure random UInt8List of [length] bytes
+Uint8List randomBytes(int length) {
+  final Random random = Random.secure();
+
+  final Uint8List randomBytesArray = Uint8List.fromList(
+    List<int>.generate(length, (i) => random.nextInt(256)),
```

Comment:
Instead of generating two lists, you could just create the uint8list and fill it in a for loop.